applications/DEM_application/custom_strategies/strategies/explicit_solver_strategy.cpp

```diff
@@ -1142,15 +1179,33 @@ namespace Kratos {
                         rigid_body_element.GetGeometry()[0].Set(DEMFlags::FIXED_ANG_VEL_Y, true);
                         rigid_body_element.GetGeometry()[0].Set(DEMFlags::FIXED_ANG_VEL_Z, true);
                     }
-                }
-            }
 
-            if (submp.Has(EXTERNAL_APPLIED_FORCE)) { // JIG: Backward compatibility, it should be removed in the future
-                rigid_body_element.GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE) = submp[EXTERNAL_APPLIED_FORCE];
-            }
+                    if (submp.Has(TABLE_NUMBER_FORCE_X)) {
+                        const int table_number = submp[TABLE_NUMBER_FORCE_X];
+                        rigid_body_element.GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE)[0] = fem_model_part.GetTable(table_number).GetValue(time);
+                    }
+                    if (submp.Has(TABLE_NUMBER_FORCE_Y)) {
+                        const int table_number = submp[TABLE_NUMBER_FORCE_Y];
+                        rigid_body_element.GetGeometry()[0].FastGetSolutionStepValue(EXTERNAL_APPLIED_FORCE)[1] = fem_model_part.GetTable(table_number).GetValue(time);
```

Comment:
Looks ok but wouldn't it be better organized if the table was defined inside the submp?

Reply:
Good point @farrufat-cimne, thanks!